lib/AST/ArchetypeBuilder.cpp

```diff
@@ -2028,7 +2028,10 @@ Type ArchetypeBuilder::mapTypeIntoContext(Module *M,
   if (!canType->hasTypeParameter())
     return type;
 
-  assert(genericParams && "dependent type in non-generic context");
+  if (!genericParams) {
+    // Failed: dependent type in non-generic context.
+    return Type();
+  }
```

Comment:
The problem then is we have an enum with a raw type that's generic, but no generic parameter list.

This means some other part of the compiler upstream of you constructed an invalid declaration. If you figure out what's going on and fix it there, you might be able to squash a few more compiler crashers.

Or, you can just XFAIL the compiler crasher for now. Having one or two regressions isn't the end of the world, especially since you improved matters in other respects.